### PR TITLE
update(FilePreview): Avoid open file if it is already opened

### PR DESCRIPTION
**Pull request by lgmarchi**
### 1. Avoid to open file preview if that file is already opened, and show a notification 

https://user-images.githubusercontent.com/63157656/234697044-b8117811-ab32-4aef-898e-8185c5394c0a.mov

### 2. Show notification if file does not have thumbnail 

https://user-images.githubusercontent.com/63157656/234697208-98ac0d37-d47a-4060-b991-b422b5aa8986.mov

- 

### Which issue(s) this PR fixes 🔨

- Resolve #

### CODE REVIEW

**Comment by stavares843** [comment]
Captura de ecrã 2023-04-26, às 22 43 41

File: ui/src/layouts/storage.rs [hunk 1]
```diff
@@ -667,6 +668,7 @@ pub fn FilesLayout(cx: Scope<Props>) -> Element {
                             let file2 = file.clone();
                             let file3 = file.clone();
                             let key = file.id();
+                            let file_id = file.id().clone();
```
Comment (dariusc93) [inline ui/src/layouts/storage.rs]:
```suggestion
                            let file_id = file.id();
```

File: ui/src/layouts/storage.rs [hunk 2]
```diff
@@ -717,7 +719,34 @@ pub fn FilesLayout(cx: Scope<Props>) -> Element {
                                             aria_label: file.name(),
                                             with_rename: *is_renaming_map.read() == Some(key),
                                             onpress: move |_| {
-                                                let key = Uuid::new_v4();
+                                                let key = file_id.clone();
```
Comment (dariusc93) [inline ui/src/layouts/storage.rs]:
```suggestion
                                                let key = file_id;
```

**Review by dariusc93 — COMMENTED** [review]
`Uuid` implements `Copy` so cloning is not needed. If its behind a reference, you can dereference it instead

**Review by phillsatellite — APPROVED** [review]

**Review by dariusc93 — APPROVED** [review]

**Pull request merged**